packages/react/error-boundary/src/ErrorBoundaryGroup.tsx

```diff
@@ -48,17 +48,29 @@ export const ErrorBoundaryGroup = ({
     if (isMounted && !blockOutsideRef.current) {
       reset();
     }
-  }, [group.resetKey, isMounted, reset]);
+  }, [group?.resetKey, isMounted, reset]);
 
   const value = useMemo(() => ({ reset, resetKey }), [reset, resetKey]);
 
   return <ErrorBoundaryGroupContext.Provider value={value}>{children}</ErrorBoundaryGroupContext.Provider>;
 };
 
 export const useErrorBoundaryGroup = () => {
-  const { reset } = useContext(ErrorBoundaryGroupContext);
+  const group = useContext(ErrorBoundaryGroupContext);
+
+  if (group === undefined) {
+    throw new Error('useErrorBoundaryGroup: ErrorBoundaryGroup is required in parent');
+  }
```

Comment:
In [AsyncBoundaryContext](https://github.com/toss/slash/blob/6bb1eb5856c4ba231a9d418d2f477721be4aae63/packages/react/async-boundary/src/provider/AsyncBoundaryContext.tsx#L26), it is supposed to return the default value if there is no provider, so I wonder if it is a better way to generate an error. 

Do you think the method in AsyncBoundaryContext also needs to be improved?

Reply:
It'll be better in my opinion. And you?

I think before this Pull Request even If no parent boundary, this hook's returntype can make developer misunderstand as reset will work without Boundary. so I fixed it.

Reply:
> It'll be better in my opinion. And you?

That makes two of us. (I will make PR to AsyncBoundaryContext 🚀)

But, this hook`s return type, after this PR, also type like this

<img width="385" alt="image" src="https://github.com/toss/slash/assets/57122180/b556e7df-4683-4277-b4e4-16a50319f64f">

So, why don`t you add JSDoc for remindering to use Provider as parent.

As the `useContext` is like that

<img width="515" alt="image" src="https://github.com/toss/slash/assets/57122180/ae7be9be-a5aa-497e-a5bb-7cc3958d1bc9">

> as given by `the nearest context provider` for the given context.

Reply:
> But, this hook`s return type, after this PR, also type like this

Same type. but before this Pull Request, if no ErrorBoundaryGroup in parent, errorBoundaryGroup.reset will be () => {}, just empty function. I want to remove this case. If we use errorBoundaryGroup.reset function, I want to make developers to be confident that ErrorBoundaryGroup will be reset. so i did this.

anyway, Well informing developers with JSDoc seems like a good idea.

If this change is evaluate as not proper for slash maintainer, I want to suggest `undefined | {reset: () => void}` for ReturnType<typeof useErrorBoundaryGroup> to clarify that reset function can be useless when no ErrorBoundaryGroup in parent.

So first, I think we need to wait review first. when there is decision, I want to add jsdoc

Reply:
> I want to make developers to be confident that ErrorBoundaryGroup will be reset. so i did this.

I totally agree about this PR wants to do and agree throw error is better than just call empty default function called. 

Thank you for received my opinion 👍

Reply:
As you said, using `useErrorBoundaryGroup` where `ErrorBoundary` is not present is nonsensical. I think `throw`ing in advance makes more sense.

Reply:
> > It'll be better in my opinion. And you?
> 
> That makes two of us. (I will make PR to AsyncBoundaryContext 🚀 after this PR merged) It is better way of case for developer using this hook who don`t know what is wrong.
> 
> > this hook's returntype can make developer misunderstand
> 
> But, this hook`s return type, after this PR, also type like this
> 
> <img alt="image" width="385" src="https://user-images.githubusercontent.com/57122180/246391745-b556e7df-4683-4277-b4e4-16a50319f64f.png">
> So, why don`t you add JSDoc for remindering to use Provider as parent.
> 
> As the `useContext` is like that
> 
> <img alt="image" width="515" src="https://user-images.githubusercontent.com/57122180/246392303-ae7be9be-a5aa-497e-a5bb-7cc3958d1bc9.png">
> > as given by `the nearest context provider` for the given context.

@minsoo-web 
It's your time to make PR like this for AsyncBoundary as your comment. I just want to alert you.